File: src/main/java/streams/SumNumbers.java
package streams;

import java.util.List;

public class SumNumbers {
    public static int getTotal(List<Integer> numbers) {
        return numbers.stream()
                .filter(n -> n % 3 == 0)
                .map(n -> n * 2)
                // .reduce(0, (acc, val) -> acc + val);
                .reduce(0, Integer::sum);
    }
}

File: src/main/kotlin/sequences/sum_numbers.kt
package sequences

fun getTotal(nums: List<Int>) =
        nums.filter { it % 3 == 0 }
                .map { it * 2 }
                // .sum()
                .fold(0) { acc, i -> acc + i }

fun getTotalUsingSequences(nums: List<Int>) =
        nums.asSequence()
                .filter { it % 3 == 0 }
                .map { it * 2 }
                .sum()